eventsource/format-field-id-null.htm

```diff
@@ -0,0 +1,33 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <title>EventSource: U+0000 in id field</title>
+    <script src="/resources/testharness.js"></script>
+    <script src="/resources/testharnessreport.js"></script>
+  </head>
+  <body>
+    <div id="log"></div>
+    <script>
+      var test = async_test()
+      test.step(function() {
+        var source = new EventSource("resources/last-event-id.py?idvalue=%00%00"),
+            seenhello = false
+        source.onmessage = function(e) {
+          test.step(function() {
+            if(e.data == "hello" && !seenhello) {
+              seenhello = true
+              assert_equals(e.lastEventId, "\0\0")
```

Comment:
I am trying to get this test to pass in WebKit via https://bugs.webkit.org/show_bug.cgi?id=175178. However, I get the empty string here instead of "\0\0". Would you mind clarifying why we expect lastEventId to contain NUL here (I thought this is what we were trying to avoid).

Disclaimer: I am very unfamiliar with EventSource so this may well be a silly question :)